src/price.rs

```diff
@@ -26,4 +27,48 @@ impl Contract {
             panic!("Market for token: {} was not found", token);
         })
     }
+
+    pub fn get_liquidation_price(
+        &self,
+        sell_token_amount: U128,
+        sell_token_price: U128,
+        buy_token_price: U128,
+        leverage: U128,
+        borrow_fee: U128,
+        swap_fee: U128,
+    ) -> WBigDecimal {
+
+        let collateral_usd = BigDecimal::from(sell_token_amount) * BigDecimal::from(sell_token_price);
+        
+        let buy_amount = collateral_usd / BigDecimal::from(buy_token_price);
+
+        let borrow_amount = collateral_usd * BigDecimal::from(leverage);
+
+        (BigDecimal::from(buy_token_price) - (collateral_usd - BigDecimal::from(borrow_fee) * borrow_amount) / buy_amount + BigDecimal::from(sell_token_amount) * BigDecimal::from(swap_fee)).into()
+    }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_liquidation_price() {
```

Comment:
@vlad-sundunchan, please use our calculations here https://docs.google.com/spreadsheets/d/194ero95KPezi2yvlq5I-BK5_Bowj2Vz75J6EOQzZq8s/edit?pli=1#gid=1392357925
to provide tests for each case